### PR TITLE
Switch controller-manager to logr (5)

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area logging dev-productivity
/kind enhancement

**What this PR does / why we need it**:

Continuation of https://github.com/gardener/gardener/pull/5488
Migrate controller-manager shoot controller to logr.
This was the last one in controller-manager, meaning it is fully migrated now.

**Which issue(s) this PR fixes**:
Part of #4251

**Special notes for your reviewer**:

We have to keep instantiating `logger.Logger` in controller-manager for now, as there are a few minor usages in helper funcs shared with gardenlet. 
We can drop this, once they have been switched as well.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/controllermanager/controller/shoot/shoot_maintenance_control.go [hunk 1]
```diff
@@ -69,94 +73,92 @@ func (c *Controller) shootMaintenanceDelete(obj interface{}) {
...
 		return reconcile.Result{}, nil
 	}
 
-	requeueAfter := requeueAfterDuration(shoot)
+	requeueAfter, nextMaintenance := requeueAfterDuration(shoot)
 
 	if !mustMaintainNow(shoot) {
-		logger.Logger.Infof("[SHOOT MAINTENANCE] %s/%s - skipping because Shoot must not be maintained now.", shoot.Namespace, shoot.Name)
+		log.V(1).Info("Skipping Shoot because it doesn't need to be maintained now")
+		log.V(1).Info("Scheduled next maintenance for Shoot", "duration", requeueAfter.Round(time.Minute), "nextMaintenance", nextMaintenance.Round(time.Minute))
```
Comment (rfranzke) [inline pkg/controllermanager/controller/shoot/shoot_maintenance_control.go:118]: Why `V(1)`? This was `Info` before

File: pkg/controllermanager/controller/shoot/shoot_retry_control.go [hunk 2]
```diff
@@ -52,63 +54,56 @@ func (c *Controller) shootRetryUpdate(oldObj, newObj interface{}) {
...
 		return reconcile.Result{}, nil
 	}
 
 	mustRetry, requeueAfter := mustRetryNow(shoot, *r.config.RetryPeriod, r.config.RetryJitterPeriod)
 	if !mustRetry {
-		shootLogger.Infof("[SHOOT RETRY] Scheduled retry in %s", requeueAfter.Round(time.Second))
+		if requeueAfter == 0 {
+			return reconcile.Result{}, nil
+		}
+		log.V(1).Info("Scheduling retry for Shoot", "requeueAfter", requeueAfter.Round(time.Second))
```
Comment (rfranzke) [inline pkg/controllermanager/controller/shoot/shoot_retry_control.go:98]: Same question here

File: pkg/controllermanager/controller/shoot/shoot_retry_control.go [hunk 3]
```diff
@@ -52,63 +54,56 @@ func (c *Controller) shootRetryUpdate(oldObj, newObj interface{}) {
...
+	log.Info("Retrying failed Shoot")
 
-	shootCopy := shoot.DeepCopy()
+	patch := client.MergeFrom(shoot.DeepCopy())
 	metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationRetry)
-	if err := r.gardenClient.Patch(ctx, shoot, client.MergeFrom(shootCopy)); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	shootLogger.Info("[SHOOT RETRY] Shoot was successfully retried")
```
Comment (rfranzke) [inline pkg/controllermanager/controller/shoot/shoot_retry_control.go:109]: Why is this log statement removed?

File: pkg/controllermanager/controller/shoot/shoot_maintenance_control.go (hunk 1 above)
Comment (timebertt) [reply pkg/controllermanager/controller/shoot/shoot_maintenance_control.go:118]: I think, logging the scheduled next maintenance for every shoot is pretty noisy. Hence, increased the log level to debug

File: pkg/controllermanager/controller/shoot/shoot_retry_control.go (hunk 2 above)
Comment (timebertt) [reply pkg/controllermanager/controller/shoot/shoot_retry_control.go:98]: Same answer here ;)

File: pkg/controllermanager/controller/shoot/shoot_retry_control.go (hunk 3 above)
Comment (timebertt) [reply pkg/controllermanager/controller/shoot/shoot_retry_control.go:109]: Usually, we don't log about successful calls. The absence of errors will implicitly state that it was successfully retried.
Also, we already log what we are about to do earlier, so I found this log to be duplicated in a sense.

**Pull request merged**